addin-fixed-weights/program/src/processor.rs

```diff
@@ -89,6 +98,45 @@ pub fn process_setup_voter_weight_record(
     Ok(())
 }
 
+/// Processes Set Delegate instruction
+pub fn process_set_partial_voting(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    vote_percentage: u16,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let realm_info = next_account_info(account_info_iter)?; // 0
+    let governing_token_mint_info = next_account_info(account_info_iter)?; // 1
+    let governing_token_owner_info = next_account_info(account_info_iter)?; // 2
+    let voter_weight_record_info = next_account_info(account_info_iter)?; // 3
+
+    if !governing_token_owner_info.is_signer {
+        return Err(VoterWeightAddinError::MissingRequiredSigner.into());
+    }
+
+    let mut voter_weight_record = get_account_data::<VoterWeightRecord>(program_id, voter_weight_record_info)?;
```

Comment:
See on https://github.com/neonlabsorg/neon-spl-governance/blob/main/addin-vesting/program/src/processor.rs#L472. The main idea: not only the owner, but the delegate can execute this command.

Reply:
Updated. New Authority delegate implemented.